simulator/src/main/java/simblock/node/consensus/ProofOfSpace.java

```diff
@@ -0,0 +1,68 @@
+package simblock.node.consensus;
+
+import simblock.block.Block;
+import simblock.block.ProofOfWorkBlock;
+import simblock.node.Node;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+import simblock.task.MiningTask;
+import simblock.task.WinningTask;
+
+import java.math.BigInteger;
+
+import static simblock.simulator.Main.random;
+
+public class ProofOfSpace extends AbstractConsensusAlgo{
+
+    public ProofOfSpace(){
+        for(int i = 1; i<=10; i++){
+            Timer.getSimulationTimer().putTask(new WinningTask());
+        }
+    }
+    @Override
+    public MiningTask CreateMintingTask(Node node){
```

Comment:
a node starts minting on it's current block and generates a mining task w some probability corresponding to the difficulty